File: library/src/main/java/cn/woolsen/android/uitl/SPUtils.java
package cn.woolsen.android.uitl;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.collection.SimpleArrayMap;

import java.util.Set;

import cn.woolsen.android.mvp.BaseApplication;

/**
 * create by woolsen on 19/7/12
 */
public class SPUtils {

    private SharedPreferences sp;

    private static SimpleArrayMap<String, SPUtils> map = new SimpleArrayMap<>();

    public static SPUtils get(String fileName) {
        if (!map.containsKey(fileName)) {
            map.put(fileName, new SPUtils(BaseApplication.getContext(), fileName));
        }
        return map.get(fileName);
    }

    private SPUtils(Context context, String fileName) {
        sp = context.getSharedPreferences(fileName, Context.MODE_PRIVATE);
    }
    
    public void putString(String key, String value) {
        SharedPreferences.Editor edit = sp.edit();
        edit.putString(key, value);
        edit.apply();
    }

    public void putStringSet(String key, Set<String> value) {
        SharedPreferences.Editor edit = sp.edit();
        edit.putStringSet(key, value);
        edit.apply();
    }

    public void putInt(String key, Integer value) {
        SharedPreferences.Editor edit = sp.edit();
        edit.putInt(key, value);
        edit.apply();
    }

    public void putBoolean(String key, boolean value) {
        SharedPreferences.Editor edit = sp.edit();
        edit.putBoolean(key, value);
        edit.apply();
    }

    public Set<String> getStringSet(String key, Set<String> defaultValue) {
        return sp.getStringSet(key, defaultValue);
    }

    public String getString(String key) {
        return sp.getString(key, "");
    }

    public boolean getBoolean(String key) {
        return sp.getBoolean(key, false);
    }

    public void remove(String key) {
        sp.edit().remove(key).apply();
    }

    public boolean contain(String key) {
        return sp.contains(key);
    }

    public void clear() {
        sp.edit().clear().apply();
    }
}

File: app/src/main/java/cn/ifafu/ifafu/util/SPUtils.kt
package cn.ifafu.ifafu.util

import android.content.Context
import android.content.SharedPreferences
import androidx.collection.SimpleArrayMap
import cn.ifafu.ifafu.base.BaseApplication

/**
 * SharedPreferences工具
 *
 * create by woolsen on 19/7/12
 */
class SPUtils private constructor(context: Context, fileName: String) {

    private val sp: SharedPreferences = context.getSharedPreferences(fileName, Context.MODE_PRIVATE)

    fun putString(key: String, value: String) {
        sp.edit().apply {
            putString(key, value)
        }.apply()
    }

    fun putStringSet(key: String, value: Set<String>) {
        sp.edit().apply {
            putStringSet(key, value)
        }.apply()
    }

    fun putInt(key: String, value: Int) {
        sp.edit().apply {
            putInt(key, value)
        }.apply()
    }

    fun putBoolean(key: String, value: Boolean) {
        sp.edit().apply { putBoolean(key, value)
        }.apply()
    }

    fun getStringSet(key: String): Set<String> {
        return sp.getStringSet(key, emptySet())!!
    }

    fun getString(key: String): String {
        return sp.getString(key, "")!!
    }

    fun getBoolean(key: String): Boolean {
        return sp.getBoolean(key, false)
    }

    val all: MutableMap<String, *>
        get() = sp.all

    fun remove(key: String) {
        sp.edit().apply {
            remove(key)
        }.apply()
    }

    fun contain(key: String): Boolean {
        return sp.contains(key)
    }

    fun clear() {
        sp.edit().clear().apply()
    }

    companion object {
        private val map = SimpleArrayMap<String, SPUtils>()
        @JvmStatic
        operator fun get(fileName: String): SPUtils {
            return map[fileName] ?: kotlin.run {
                val sp = SPUtils(BaseApplication.appContext, fileName)
                map.put(fileName, sp)
                sp
            }
        }
    }

}